apis/configwatch.go

```diff
@@ -181,3 +205,20 @@ func MonitorTLS(start time.Time, expiry time.Time, conf *config.Config) {
 		}
 	}
 }
+
+func (s *Server) getClientValidator(helloInfo *tls.ClientHelloInfo) func([][]byte, [][]*x509.Certificate) error {
+
+	log.Debug().Msgf("Inside get client validator - %v", helloInfo.Conn.RemoteAddr())
+	return func(rawCerts [][]byte, verifiedChains [][]*x509.Certificate) error {
+		//added DNSName
+		opts := x509.VerifyOptions{
+			Roots:         s.CaCertPool,
+			CurrentTime:   time.Now(),
+			Intermediates: x509.NewCertPool(),
+			KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
+			DNSName:       strings.Split(helloInfo.Conn.RemoteAddr().String(), ":")[0],
```

Comment:
I believe this will have to be a list of DNS Names we match on.  This looks like it will return the IP address.